src/main/java/io/cdap/plugin/salesforce/parser/SalesforceQueryVisitor.java

```diff
@@ -158,6 +158,30 @@ public SObjectDescriptor visitStatement(SOQLParser.StatementContext ctx) {
       return new SObjectDescriptor.FieldDescriptor(nameParts, alias, SalesforceFunctionType.NONE);
     }
 
+    @Override
+    public SObjectDescriptor.FieldDescriptor visitSelectKeyField(SOQLParser.SelectKeyFieldContext ctx) {
+      List<String> nameParts = ctx.selectKey().stream()
+        .map(RuleContext::getText)
+        .collect(Collectors.toList());
+
+      if (nameParts.isEmpty()) {
+        throw new SOQLParsingException("Invalid column name: " + ctx.getText());
+      }
+
+      if (nameParts.size() > 1) {
+        // if field contains alias, remove it to get actual field name
+        // Example:
+        // select Opportunity.Name from Opportunity -> Name
+        // select o.Name from Opportunity o -> Name
+        String topParent = nameParts.get(0);
+        if (topParent.equalsIgnoreCase(objectName) || topParent.equals(objectAlias)) {
```

Comment:
why do we ignore case when checking equality with objectName, but not with objectAlias?